packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -8,6 +8,67 @@ import { getSortKeyFields } from './schema';
 import { HasManyDirectiveConfiguration } from './types';
 import { getConnectionAttributeName, getObjectPrimaryKey } from './utils';
 
+/**
+ * Creates a GSI on the table of the `relatedType` based on the config's `references` / `referenceNodes`
+ *
+ * @remarks
+ * This method sets the `indexName` property of the `config` to the GSI name created on the
+ * table of the `relatedType`
+ *
+ * Preconditions: `config.references >= 1` and `config.referenceNodes >= 1`
+ *
+ * @param config The `HasManyDirectiveConfiguration` for DDB references.
+ * @param ctx The `TransformerContextProvider` for DDB references.
+ */
+export const updateTableForReferencesConnection = (
+  config: HasManyDirectiveConfiguration, // TODO: Add support for HasOneDirectiveConfiguration
+  ctx: TransformerContextProvider,
+): void => {
+  const { field, referenceNodes, indexName: incomingIndexName, object, references, relatedType } = config;
+
+  if (incomingIndexName) {
+    // TODO: log warning or throw that indexName isn't supported for DDB references
+    // Ideally validate this further up the chain.
+  }
+
+  if (references.length < 1 || referenceNodes.length < 1) {
+    throw new Error('references should not be empty here'); // TODO: better error message
+  }
+
+  const mappedObjectName = ctx.resourceHelper.getModelNameMapping(object.name.value);
+  const indexName = `gsi-${mappedObjectName}.${field.name.value}`;
+  config.indexName = indexName;
+
+  const relatedTable = getTable(ctx, relatedType);
+  const gsis = relatedTable.globalSecondaryIndexes;
+  if (gsis.some((gsi: any) => gsi.indexName === indexName)) {
+    // TODO: In the existing `fields` based implementation, this returns.
+    // However, this is likely a schema misconfiguration in the `references`
+    // world because we don't support specifying indexName.
+    return;
+  }
+
+  const referenceNode = referenceNodes[0];
+  const partitionKeyName = referenceNode.name.value;
+  // Grabbing the type of the related field.
+  // TODO: Validate types of related field and primary's pk match
+  // -- ideally further up the chain
+  const partitionKeyType = attributeTypeFromType(referenceNode.type, ctx);
+  const respectPrimaryKeyAttributesOnConnectionField: boolean = ctx.transformParameters.respectPrimaryKeyAttributesOnConnectionField;
+
+  const sortKey = respectPrimaryKeyAttributesOnConnectionField
+    ? getConnectedSortKeyAttributeDefinitionsForImplicitHasManyObject(ctx, object, field, referenceNodes.slice(1))
```

Comment:
It seems that this method is not used correctly by the new ref relation. If you compare the last argument for both `ref` and `field`, the former passes the `referenceNodes.slice(1)` which is the related model fields, whereas the latter passes `getSortKeyFields(ctx, object)` which are the sort key fields of the primary. And [these lines of code](https://github.com/aws-amplify/amplify-category-api/pull/2362/files#diff-7e54ffdc31c20805d81c17a3e63863a1553b9911a6a00daf1e21f027f2995d84L98-L100) stays the same. I wonder if the sort key definition is expected.

Reply:
We should be careful to reuse some methods which were intended for the `field` usage, such as this one. It will also be confusing to reuse it in the `ref` scene as there is not `implicitHasMany` concept here. I would suggest to have a new method instead of reusing the old one. The essence of this method is to get the sort key definition for the auto-generated index in related model.

Reply:
You're absolutely right, great catch! I'm pushing that to a subsequent PR to keep the scope of this one as tight as possible. Removed and added a TODO in 5debc1a8e1c8e15b8f5aaaba341adf3ecfd959ac